frontend/components/tree/js/tree.service.ts

```diff
@@ -1131,10 +1138,31 @@ export class TreeService {
 
 		}
 
+		const lastNode = nodes[nodes.length - 1] ;
+		this.handleMetadata(lastNode);
+
 		return this.highlightNodes(nodes, multi, colour);
 
 	}
 
+	/**
+	 * Show metadata in the metadata panel if necessary
+	 * @param lastNode the node to show the metadata for
+	 */
+	public handleMetadata(lastNode) {
+		// Handle showing metadata in the metadata panel if necessary
+
+		if (lastNode && lastNode.meta) {
+			if (this.DocsService.state.active && !this.ViewerService.pin.pinDropMode) {
```

Comment:
Dont do this check here, do it on handleObjectSelected within Doc's service.

Tree Service should tell DocsService there is metadata available on current state. It should be up to DocsService what to do with that information (whether to display, or ignore because it's pinMode or it's inactive)

If you have to keep pinMode here because of potential circular dependencies, fine (not ideal but oh well). But it shouldn't check Doc's service active state here. (imagine if you have another service needed to call Doc Service to handle document in the future, they would all have to know to check its state first.)

Reply:
That's a good point. Doesn't cause a circular dependency so i'll put it in there.